casper/src/test/scala/coop/rchain/casper/util/rholang/InterpreterUtilTest.scala

```diff
@@ -753,81 +622,69 @@ class InterpreterUtilTest
-      val b3DeploysWithCost      = prepareDeploys(Vector("@3!(3)"), PCost(1))
-
-      /*
-       * DAG Looks like this:
-       *
-       *           b3
-       *          /  \
-       *        b1    b2
-       *         \    /
-       *         genesis
-       */
-
-      mkRuntimeManager("interpreter-util-test")
-        .use { runtimeManager =>
-          for {
-            genesis <- createGenesis[Task](deploys = genesisDeploysWithCost)
-            b1      <- createBlock[Task](Seq(genesis.blockHash), genesis, deploys = b1DeploysWithCost)
-            b2      <- createBlock[Task](Seq(genesis.blockHash), genesis, deploys = b2DeploysWithCost)
-            b3 <- createBlock[Task](
-                   Seq(b1.blockHash, b2.blockHash),
-                   genesis,
-                   deploys = b3DeploysWithCost
-                 )
-            _   <- step(runtimeManager)(0, genesis)
-            _   <- step(runtimeManager)(1, genesis)
-            _   <- step(runtimeManager)(2, genesis)
-            dag <- blockDagStorage.getRepresentation
-            blockHashes <- InterpreterUtil.findMultiParentsBlockHashesForReplay(
-                            Seq(b1, b2),
-                            dag
-                          )
-            _ = withClue("Main parent hasn't been filtered out: ") { blockHashes.size shouldBe (1) }
+  "findMultiParentsBlockHashesForReplay" should "filter out duplicate ancestors of main parent block" in withGenesis(
+    genesisContext
+  ) { implicit blockStore => implicit blockDagStorage => runtimeManager =>
+    val b1DeploysWithCost = prepareDeploys(Vector("@1!(1)"), PCost(1))
+    val b2DeploysWithCost = prepareDeploys(Vector("@2!(2)"), PCost(1))
+    val b3DeploysWithCost = prepareDeploys(Vector("@3!(3)"), PCost(1))
+
+    /*
+     * DAG Looks like this:
+     *
+     *           b3
+     *          /  \
+     *        b1    b2
+     *         \    /
+     *         genesis
+     */
 
-          } yield ()
-        }
+    for {
+      b1 <- buildBlock[Task](Seq(genesis.blockHash), deploys = b1DeploysWithCost)
+      b2 <- buildBlock[Task](Seq(genesis.blockHash), deploys = b2DeploysWithCost)
+      _ <- buildBlock[Task](
+            Seq(b1.blockHash, b2.blockHash),
+            deploys = b3DeploysWithCost
+          )
+      _           <- step(runtimeManager)(b1, genesis)
+      _           <- step(runtimeManager)(b2, genesis)
+      dag         <- blockDagStorage.getRepresentation
+      blockHashes <- InterpreterUtil.findMultiParentsBlockHashesForReplay(Seq(b1, b2), dag)
+      _           = withClue("Main parent hasn't been filtered out: ") { blockHashes.size shouldBe 1 }
+    } yield ()
   }
 
-  def step[F[_]: IndexedBlockDagStorage: BlockStore: Time: Metrics: Span: Sync](
+  def step[F[_]: BlockDagStorage: BlockStore: Time: Metrics: Span: Sync](
       runtimeManager: RuntimeManager[F]
-  )(index: Int, genesis: BlockMessage): F[Unit] =
+  )(b1: BlockMessage, genesis: BlockMessage): F[Unit] =
```

Comment:
Nitpick: rename to b

Reply:
Done in a later commit

Reply:
See `Deduplicate the `step` method, moving it to BlockGenerator`